util/network-devp2p/src/host.rs

```diff
@@ -305,6 +344,34 @@ impl Host {
 		debug!(target: "network", "Listening at {:?}", listen_address);
 		let udp_port = config.udp_port.unwrap_or_else(|| listen_address.port());
 		let local_endpoint = NodeEndpoint { address: listen_address, udp_port };
+		let enr_key = enr::DefaultKey::Secp256k1(secp256k1::SecretKey::parse_slice((**keys.secret()).as_bytes()).expect("rust-secp256k1 keychain is always valid; qed"));
+		let enr = config.config_path.as_ref().and_then(|p| load::<Enr>(Path::new(&p))).filter(|enr| {
```

Comment:
Can we extract this filter to standalon function? Would be easier to understand and wouldn't pollute the code here so much.